app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/RecentSyncedTabFeature.kt

```diff
@@ -87,22 +97,37 @@ class RecentSyncedTabFeature(
      */
     override fun stopLoading() {
         if (lastSyncedTab == null) {
-            store.dispatch(AppAction.RecentSyncedTabStateChange(RecentSyncedTabState.None))
+            appStore.dispatch(AppAction.RecentSyncedTabStateChange(RecentSyncedTabState.None))
         }
     }
 
     override fun onError(error: SyncedTabsView.ErrorType) {
-        if (store.state.recentSyncedTabState == RecentSyncedTabState.Loading) {
-            store.dispatch(AppAction.RecentSyncedTabStateChange(RecentSyncedTabState.None))
+        if (appStore.state.recentSyncedTabState == RecentSyncedTabState.Loading) {
+            appStore.dispatch(AppAction.RecentSyncedTabStateChange(RecentSyncedTabState.None))
         }
     }
 
     override fun start() {
         syncedTabsFeature.start()
+        statusScope = syncStore.flowScoped { flow ->
```

Comment:
`flowScoped` is based off of `MainScoped`, so you might not need to keep the coroutine as a local variable in this instance, since it'll cancel the job in `onDestroy`, if that lifecycle works as a substitute for `stop`.

Reply:
I actually ran into some trouble with these coroutines, and so did a fair bit of investigation that led to this implementation. I'm not sure it's necessarily the best or in style with other examples in the codebase, but I'll break some of it down and if you have suggestions or examples I'm happy to update it.

- Just as a point of clarity: `MainScope` is not tied to lifecycles at all. It just creates a new `CoroutineScope` on the main thread. This could even cause objects to survive longer than anticipated, as I think any running coroutines in the created scope would keep an object from being collected.

- `flowScoped` exposes an optional `owner: LifecycleOwner` parameter which can achieve the behavior you described in your comment. There's additional code in the `Store.flow` extension that it uses to cancel the flow when a lifecycle moves into the `Destroyed` state. 

- _However_, `RecentSyncedTabFeature` is currently set using `ViewBoundFeatureWrapper`, which was in keeping with existing code in the `HomeFragment`. The whole purpose of `ViewBoundFeatureWrapper` is to keep a strong reference to objects across destructive view lifecycle events. This can prevent expensive operations like reloading a bunch of synced tabs. That said, the purpose of `ViewBoundFeatureWrapper` is the thing I understand least here so maybe it's not even necessary in this case.

- Normally in Fragments I've seen it suggested to use `viewLifecycleOwner` for automatic cancellation and re-collection of flows.

All of the above combines to mean that the normal lifecycle owner I'd use here for collecting these flows (viewLifecycleOwner) is destroyed but the `Feature` outlives it. The flow collection will stop as soon as view is destroyed and not started again. This was of manually handling the scope lifecycles prevents all that.

Reply:
gotcha gotcha. I appreciate the thorough write-up for all of the context! I'm sure you've gone through several iterations of stuff with this overall effort.

Reply:
Regarding the `ViewBoundFeatureWrapper` it was added in https://github.com/mozilla-mobile/android-components/issues/1943 to fix leaks or even crashes from features that updates various Fragment Views and for that receive them as parameters.

It's a simple convenience to ensure that when the Fragment that creates a certain feature referencing a View is added to the backstack - and so the layout and Views are destroyed - the feature is also destroyed.

As a general rule of thumb if a fragment needs a reference to a View then it should be wrapped in a `ViewBoundFeatureWrapper`.

Reply:
@Mugurell thanks for the clarity around `ViewBoundFeatureWrapper`. `RecentSyncedTabFeature` doesn't need a reference to a view, so it should be safe to stop wrapping it based on your rule of thumb. I'm still not sure I fully understand the case where `ViewBoundFeatureWrapper` prevents leaks, however. Would it be a scenario like this?

```
class MyFragment : Fragment() {
    val feature = MyFeature()

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        feature.view = view
    }
}
```

So is `ViewBoundFeatureWrapper` basically equivalent to the following?
```
class MyFragment : Fragment() {
    val feature: MyFeature? = null

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        feature = MyFeature(view)
    }

    override fun onDestroyView() {
        feature = null
    }
}
```

I've added a separate commit that simplifies the feature's logic in handling the coroutine scopes. To my understanding these updates aren't at risk of creating leaks, but I would be interesting in getting a second opinion there 😂. I'll squash the second commit if it looks safe.

@MozillaNoah as an aside, the A-C patch has landed!

Reply:
Seeing many features in `HomeFragment` being used inside of the said `ViewBoundFeatureWrapper` though they don't need a View and have injected a reference to the root layout for the required View argument.
While not optimal (based on the above) it should't cause any issues with them being initialized in `onCreateView` and then destroyed in `onDestroyView` (by the wrapper).
Though them being a `LifecycleAwareFeature` that only works while the fragment is in foreground would be enough.